### PR TITLE
hid: Tweaks, Analog Sticks

### PR DESCRIPTION
**Pull request by shinyquagsire23**
Update all controller timestamps always, update mouse+keyboard states, update unknown section states, add analog stick support.

### CODE REVIEW

File: src/core/hle/service/hid/hid.h [hunk 1]
```diff
@@ -48,6 +48,13 @@ enum ControllerConnectionState {
     ConnectionState_Wired = 1 << 1,
 };
 
+enum ControllerJoystick {
+    Joystick_Left = 0,
+    Joystick_Right = 1,
+
+    Joystick_NumSticks = 2,
```
Comment (bunnei) [inline src/core/hle/service/hid/hid.h]: nit: unused

File: src/core/hle/service/hid/hid.cpp [hunk 2]
```diff
@@ -79,61 +79,65 @@ class IAppletResource final : public ServiceFramework<IAppletResource> {
...
+                // HID shared memory stores the state of the past 17 samples in a circlular buffer,
+                // each with a timestamp in number of samples since boot.
+                layout.header.timestamp_ticks = CoreTiming::GetTicks();
+                layout.header.latest_entry = (layout.header.latest_entry + 1) % HID_NUM_ENTRIES;
+
+                ControllerInputEntry& entry = layout.entries[layout.header.latest_entry];
+                entry.connection_state = ConnectionState_Connected | ConnectionState_Wired;
+                entry.timestamp++;
+                entry
+                    .timestamp_2++; // TODO(shinyquagsire23): Is this always identical to timestamp?
```
Comment (Subv) [inline src/core/hle/service/hid/hid.cpp]: Can you move the comment to before the increment so clang-format doesn't do this idiocy?

File: src/core/hle/service/hid/hid.cpp [hunk 3]
```diff
@@ -79,61 +82,71 @@ class IAppletResource final : public ServiceFramework<IAppletResource> {
...
-            state.rstick_down.Assign(buttons[RStick_Down - BUTTON_HID_BEGIN]->GetStatus());
-
-            state.sl.Assign(buttons[SL - BUTTON_HID_BEGIN]->GetStatus());
-            state.sr.Assign(buttons[SR - BUTTON_HID_BEGIN]->GetStatus());
-
-            // TODO(shinyquagsire23): Analog stick vals
-
-            // TODO(shinyquagsire23): Update pad info proper, (circular buffers, timestamps,
-            // layouts)
+        for (int controller = 0; controller < mem.controllers.size(); controller++) {
```
Comment (lioncash) [inline src/core/hle/service/hid/hid.cpp]: `int` should be a `size_t` here.

File: src/core/hle/service/hid/hid.cpp [hunk 4]
```diff
@@ -79,61 +82,71 @@ class IAppletResource final : public ServiceFramework<IAppletResource> {
...
+                state.rstick_left.Assign(buttons[RStick_Left - BUTTON_HID_BEGIN]->GetStatus());
+                state.rstick_up.Assign(buttons[RStick_Up - BUTTON_HID_BEGIN]->GetStatus());
+                state.rstick_right.Assign(buttons[RStick_Right - BUTTON_HID_BEGIN]->GetStatus());
+                state.rstick_down.Assign(buttons[RStick_Down - BUTTON_HID_BEGIN]->GetStatus());
+
+                state.sl.Assign(buttons[SL - BUTTON_HID_BEGIN]->GetStatus());
+                state.sr.Assign(buttons[SR - BUTTON_HID_BEGIN]->GetStatus());
+
+                float stick_l_x_f, stick_l_y_f, stick_r_x_f, stick_r_y_f;
+                std::tie(stick_l_x_f, stick_l_y_f) = sticks[Joystick_Left]->GetStatus();
```
Comment (lioncash) [inline src/core/hle/service/hid/hid.cpp]: you can do:

```cpp
const auto [stick_l_x_f, stick_l_y_f] = sticks[Joystick_Left]->GetStatus();
const auto [stick_r_x_f, stick_r_y_f] = sticks[Joystick_Right]->GetStatus();
```
to assign on the spot.

File: src/core/hle/service/hid/hid.cpp [hunk 5]
```diff
@@ -151,6 +164,71 @@ class IAppletResource final : public ServiceFramework<IAppletResource> {
...
+        keyboard.header.timestamp_ticks = timestamp;
+        keyboard.header.num_entries = keyboard.entries.size();
+        keyboard.header.latest_entry = last_keyboard_entry;
+        keyboard.header.max_entry_index = keyboard.entries.size();
+
+        keyboard.entries[curr_keyboard_entry].timestamp = keyboard_sample_counter;
+        keyboard.entries[curr_keyboard_entry].timestamp_2 = keyboard_sample_counter;
+
+        // TODO(shinyquagsire23): Figure out what any of these are
+        for (int i = 0; i < mem.unk_input_1.size(); i++) {
```
Comment (lioncash) [inline src/core/hle/service/hid/hid.cpp]: ditto about `size_t`

File: src/core/hle/service/hid/hid.cpp [hunk 6]
```diff
@@ -151,6 +164,71 @@ class IAppletResource final : public ServiceFramework<IAppletResource> {
...
+            input.header.timestamp_ticks = timestamp;
+            input.header.num_entries = input.entries.size();
+            input.header.latest_entry = last_input_entry;
+            input.header.max_entry_index = input.entries.size();
+
+            input.entries[curr_input_entry].timestamp = input_sample_counter;
+            input.entries[curr_input_entry].timestamp_2 = input_sample_counter;
+        }
+
+        for (int i = 0; i < mem.unk_input_2.size(); i++) {
```
Comment (lioncash) [inline src/core/hle/service/hid/hid.cpp]: ditto

**Pull request merged**